src/HtmlUtils.js

```diff
@@ -92,31 +102,64 @@ class Highlighter {
         }
         else {
             // no more highlights to be found, just return the unhighlighted string
-            return [this._createSpan(safeSnippet, false)];
+            return [this._processSnippet(safeSnippet, false)];
         }
     }
+}
+
+class HtmlHighlighter extends BaseHighlighter {
+    /* highlight the given snippet if required
+     *
+     * snippet: content of the span; must have been sanitised
+     * highlight: true to highlight as a search match
+     *
+     * returns an HTML string
+     */
+    _processSnippet(snippet, highlight) {
+        if (!highlight) {
+            // nothing required here
+            return snippet;
+        }
+
+        var span = "<span class=\""+this.highlightClass+"\">"
+            + snippet + "</span>";
+
+        if (this.highlightLink) {
+            span = "<a href=\""+encodeURI(this.highlightLink)+"\">"
+                +span+"</a>";
+        }
+        return span;
+    }
+}
+
+class TextHighlighter extends BaseHighlighter {
+    constructor(highlightClass, highlightLink) {
+        super(highlightClass, highlightLink);
+        this._key = 0;
+    }
 
     /* create a <span> node to hold the given content
      *
-     * spanBody: content of the span. If html, must have been sanitised
+     * snippet: content of the span
      * highlight: true to highlight as a search match
+     *
+     * returns a React node
      */
-    _createSpan(spanBody, highlight) {
+    _processSnippet(snippet, highlight) {
         var spanProps = {
             key: this._key++,
         };
 
         if (highlight) {
-            spanProps.onClick = this.onHighlightClick;
             spanProps.className = this.highlightClass;
         }
 
-        if (this.html) {
-            return (<span {...spanProps} dangerouslySetInnerHTML={{ __html: spanBody }} />);
-        }
-        else {
-            return (<span {...spanProps}>{ spanBody }</span>);
+        var node = <span {...spanProps}>{ snippet }</span>;
```

Comment:
I don't see why we can drop the dangerouslySetInnerHTML here - doesn't this result in the html tags like `<span/>` being escaped by react when the snippet is rendered?